src/partition/repart.c

```diff
@@ -3018,7 +3018,11 @@ static int context_discard_gap_after(Context *context, Partition *p) {
         if (p)
                 gap = p->offset + p->new_size;
         else
-                gap = context->start;
+                /* The context start gets rounded up to grain_size, however
+                 * existing partitions may be before that so ensure the gap
+                 * starts at the first actually usable lba
+                 */
+                gap = fdisk_get_first_lba(context->fdisk_context) * context->sector_size;
```

Comment:
makes sense, but the same problem exists at the end of the disk I guess. i.e. can you make a similar change further down, where we do:

```c
if (next == UINT64_MAX) {
        next = context->end;
        …
```

i.e. change that too, to go until fdisk_get_last_lba()+1.